src/main/java/org/mule/tools/maven/plugin/mule/arm/ArmApi.java

```diff
@@ -60,25 +60,44 @@ public Application getApplicationStatus(int applicationId)
         return get(uri, APPLICATIONS + "/" + applicationId, Application.class);
     }
 
-    public String undeployApplication(int applicationId)
+    public Response undeployApplication(int applicationId)
```

Comment:
This class should abstract the client from the underlying connection, returning a Response forces the client to deal with it.